### PR TITLE
Add TinyImageNet fetcher and fixes for tests.

### PR DESCRIPTION
**Pull request by crockpotveggies**
## What changes were proposed in this pull request?

Adds a tiny image net dataset fetcher and fixes some issues with tests caused by public API changes in CSVRecordReader.

## How was this patch tested?

Manual tests. Working with @raver119 on how to best proceed for his purposes.

### CODE REVIEW

File: deeplearning4j-core/src/main/java/org/deeplearning4j/datasets/fetchers/TinyImageNetFetcher.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+package org.deeplearning4j.datasets.fetchers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+
+public class TinyImageNetFetcher extends CacheableExtractableDataSetFetcher {
```
Comment (huitseeker) [inline deeplearning4j-core/src/main/java/org/deeplearning4j/datasets/fetchers/TinyImageNetFetcher.java]: You'll want to JavaDoc what this dataset is and link to http://cs231n.stanford.edu/project.html

**Comment by maxpumperla** [comment]
@crockpotveggies do you quickly need help with this?

**Comment by crockpotveggies** [comment]
@maxpumperla thanks for asking, not yet unless you have some specific ideas in mind?

**Comment by maxpumperla** [comment]
sorry @crockpotveggies, from huitseeker's review it seemed it was only the javadoc left to change. I thought you might have either forgotten about this or didn't have the capacity, so I offered help.

What exactly is left here? Maybe we can discuss (in order to avoid PRs that remain open for too long).

**Comment by crockpotveggies** [comment]
Goal is to add a data set iterator that can be passed to a network. Let's
chat on Gitter

**Pull request merged**